homeassistant/components/systemmonitor/sensor.py

```diff
@@ -76,17 +77,164 @@
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
 
+def get_disk_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return disk use percent."""
+    disk = cast(sdiskusage, entity.coordinator.data)
+    return disk.percent
+
+
+def get_disk_use(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return disk use."""
+    disk = cast(sdiskusage, entity.coordinator.data)
+    return round(disk.used / 1024**3, 1)
+
+
+def get_disk_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return disk free."""
+    disk = cast(sdiskusage, entity.coordinator.data)
+    return round(disk.free / 1024**3, 1)
+
+
+def get_memory_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory use percent."""
+    virtual_memory = cast(svmem, entity.coordinator.data)
+    return virtual_memory.percent
+
+
+def get_memory_use(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory use."""
+    virtual_memory = cast(svmem, entity.coordinator.data)
+    return round((virtual_memory.total - virtual_memory.available) / 1024**2, 1)
+
+
+def get_memory_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory free."""
+    memory = cast(svmem, entity.coordinator.data)
+    return round(memory.available / 1024**2, 1)
+
+
+def get_swap_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap use percent."""
+    swap = cast(sswap, entity.coordinator.data)
+    return swap.percent
+
+
+def get_swap_use(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap use."""
+    swap = cast(sswap, entity.coordinator.data)
+    return round(swap.used / 1024**2, 1)
+
+
+def get_swap_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap free."""
+    swap = cast(sswap, entity.coordinator.data)
+    return round(swap.free / 1024**2, 1)
+
+
+def get_processor_use(entity: SystemMonitorSensor[dataT]) -> int:
+    """Return processor use."""
+    return round(cast(float, entity.coordinator.data))
+
+
+def get_processor_temperature(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return processor temperature."""
+    temps = cast(dict[str, list[shwtemp]], entity.coordinator.data)
+    temp = read_cpu_temperature(temps)
+    return temp
+
+
+def get_process(entity: SystemMonitorSensor[dataT]) -> str:
+    """Return process."""
+    state = STATE_OFF
```

Comment:
Side note: An on/off state for a non binary sensor is weird.

Reply:
Agree